src/commands/__tests__/add.test.js

```diff
@@ -1,4 +1,179 @@
 // @flow
 import { add, toAddOptions } from '../add';
+import { copyFixtureIntoTempDir } from 'jest-fixtures';
+import * as path from 'path';
+import * as processes from '../../utils/processes';
+import * as yarn from '../../utils/yarn';
+import * as fs from '../../utils/fs';
+import Package from '../../Package';
+import pathExists from 'path-exists';
 
-test('bolt add');
+jest.mock('../../utils/yarn');
+jest.mock('../../utils/logger');
+
+const unsafeProcessses: any & typeof processes = processes;
+const unsafeYarn: any & typeof yarn = yarn;
+
+// Helper method to check if a dependency is installed, both in the package.json and on the fs
+async function depIsInstalled(workspaceDir: string, depName: string) {
+  const pkg = await Package.init(path.join(workspaceDir, 'package.json'));
+  const dirExists = await pathExists(
+    path.join(workspaceDir, 'node_modules', depName)
+  );
+  const depInPkgJson = pkg.getDependencyType(depName) !== null;
+
+  return dirExists && depInPkgJson;
+}
+
+// a mock yarn add function to update the pakcages's config and also node_modules dir
+async function fakeYarnAdd(pkg: Package, dependencies, type = 'dependencies') {
```

Comment:
This is the yarn.add mock I'll be using in the rest of the tests later. Just a much cleaner way of handling than what I was doing before.